### PR TITLE
Quiz submission

### PR DESCRIPTION
**Pull request by jblakeman**
Felt pretty comfortable with this material.

Learning vanilla JavaScript this past week has been fun!

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -31,7 +31,9 @@ Select all that apply:
 Explain what is a REPL, and why is it important for us as developers and help with debugging?
 
 ```text
+A REPL - Read Eval Print Loop - is an interactive language interpreter that allows you to write and execute code, line by line, seeing it's output immediately.
```
Comment (joe-gz) [inline README.md:34]: :+1:

File: README.md [hunk 2]
```diff
@@ -17,12 +17,12 @@ the square brackets (for multiple-choice questions).
 
 Select all that apply:
 ```
-[] Strings
-[] Booleans
+[x] Strings
+[x] Booleans
 [] Undefined
```
Comment (joe-gz) [inline README.md:22]: undefined is actually considered a data type!

File: README.md [hunk 3]
```diff
@@ -42,7 +44,11 @@ var foods = [ ["apple","banana","strawberry"], ["pizza","fries","hamburger"] ];
 Create a For Loop that outputs the following string for each piece of fruit in the console. "I want to eat a [fruit]"
 
 ```js
-// write code here
+foods.forEach(function(fruits) {
+    fruits.forEach(function(fruit) {
+        console.log("I want to eat a " + fruit);
+    });
+});
```
Comment (joe-gz) [inline README.md:51]: give this loop a try in the console, I'm seeing it print out both fruits and the other foods.. you actually only need one loop here, and that would be to loop through `foods[0]`

File: README.md [hunk 4]
```diff
@@ -53,7 +59,7 @@ var foods = [ ["apple","banana","strawberry"], ["pizza","fries","hamburger"] ];
 How would I go about accessing the string "pizza" in the above array?
 
 ```js
-// write code here
+foods[1][0];
```
Comment (joe-gz) [inline README.md:62]: :+1:

File: README.md [hunk 5]
```diff
@@ -64,6 +70,11 @@ Describe the rules of scope in JavaScript.
 
 Your Answer:
 ```text
+Scope is dependant on which function a variable is declared in, and whether it is declared with the `var` keyword.
+
+Variables declared with the `var` keyword have a scope limited to the function in which they are declared, and any other function declared within that function.
+
+Variables declared without the `var` keyword are global variables (they can be accessed by any other function).
```
Comment (joe-gz) [inline README.md:77]: :+1:

File: README.md [hunk 6]
```diff
@@ -75,7 +86,12 @@ the variable pizza inside your method.
 
 Your Answer:
 ```js
-// write code here
+var pizza = {
+    temperature: 70,
+    bake: function() {
+        this.temperature = 300;
+    }
+}
```
Comment (joe-gz) [inline README.md:94]: :+1:

File: README.md [hunk 7]
```diff
@@ -84,7 +100,15 @@ Define a global variable instructor and set it equal to your Squad Instructor's
...
-// write code here
+var instructor = "Joe";
+
+function localizeInstructor() {
+    var instructor = "Shmo";
+    console.log(instructor); // logs "Shmo"
+}
+
+localizeInstructor();
+console.log(instructor) // logs "Joe"
```
Comment (joe-gz) [inline README.md:111]: :+1:

File: README.md [hunk 8]
```diff
@@ -94,7 +118,24 @@ Your Answer:
...
+
+I could pass a function reference as an argument to another function:
 
+    function sayHello() {
+        return "Hello"
+    }
+    function getHello(reference) {
+        console.log(reference()); // Here the parameter reference is called and its return value is logged to the console
+    }
+    getHello(sayHello); // Logs "Hello".  No trailing parentheses means sayHello is passed as a reference
```
Comment (joe-gz) [inline README.md:138]: :+1:

File: README.md [hunk 9]
```diff
@@ -102,7 +143,13 @@ Using the object literal notation, Define an object called student and give it t
...
 Your Answer:
 ```js
-// write code here
+var student = {
+    name: "John",
+    age: 31,
+    sayHello: function() {
+        return "Hi, my name is " + this.name;
+    }
+}
```
Comment (joe-gz) [inline README.md:152]: :+1:

File: README.md [hunk 10]
```diff
@@ -113,7 +160,7 @@ Your Answer:
 
 Select all that apply:
 ```
-[] Synchronous code runs at an even pace, asynchronous code runs with uneven pacing.
+[] Synchronous code runs at an even pace, asynchronous code runs with uneven pacing. // what does the question mean by "pace"? seems too general a term to be marked as true.  "pace" usually depends on if the program is multi-threaded.
 [] Synchronous code runs all at the same time, asynchronous code runs completely randomly
-[] Synchronous code runs in order (as appears in the source), asynchronous code may run at a later time.
+[x] Synchronous code runs in order (as appears in the source), asynchronous code may run at a later time.
```
Comment (joe-gz) [inline README.md:165]: :+1:  good question above! pace literally means speed here, and you're right that it is too general to be correct :)

**Comment by joe-gz** [comment]
7/9 great job, John! just note my comments on the two above and let me know if you have any questions!

File: README.md (hunk 3 above)
Comment (jblakeman) [inline README.md:51]: Thanks, I totally missed the clear detail that pizza, fries or hamburger are not fruits and that it wasn't asking to print out all foods.  Working too quickly I guess :)

**Pull request closed without merging**